Source/scrollrt.cpp

```diff
@@ -946,38 +946,57 @@ static void scrollrt_draw(CelOutputBuffer out, int x, int y, int sx, int sy, int
 }
 
 /**
- * @brief Scale up the rendered part of the back buffer to take up the full view
+ * @brief Scale up the top left part of the buffer 2x.
  */
 static void Zoom(CelOutputBuffer out)
 {
-	int wdt = gnScreenWidth / 2;
-
-	int src_x = gnScreenWidth / 2 - 1;
-	int dst_x = gnScreenWidth - 1;
-
+	int viewport_width = out.w();
+	int viewport_offset_x = 0;
 	if (PANELS_COVER) {
 		if (chrflag || questlog) {
-			wdt >>= 1;
-			src_x -= wdt;
+			viewport_width -= SPANEL_WIDTH;
+			viewport_offset_x = SPANEL_WIDTH;
 		} else if (invflag || sbookflag) {
-			wdt >>= 1;
-			src_x -= wdt;
-			dst_x -= SPANEL_WIDTH;
+			viewport_width -= SPANEL_WIDTH;
 		}
 	}
 
-	BYTE *src = out.at(src_x, gnViewportHeight / 2 - 1);
-	BYTE *dst = out.at(dst_x, gnViewportHeight - 1);
+	// We round to even for the source width and height.
+	// If the width / height was odd, we copy just one extra pixel / row later on.
+	const int src_width = (viewport_width + 1) / 2;
+	const int doubleable_width = viewport_width / 2;
+	const int src_height = (out.h() + 1) / 2;
+	const int doubleable_height = out.h() / 2;
+
+	BYTE *src = out.at(src_width - 1, src_height - 1);
+	BYTE *dst = out.at(viewport_offset_x + viewport_width - 1, out.h() - 1);
+	const bool odd_viewport_width = (viewport_width % 2) == 1;
 
-	for (int hgt = 0; hgt < gnViewportHeight / 2; hgt++) {
-		for (int i = 0; i < wdt; i++) {
+	for (int hgt = 0; hgt < doubleable_height; hgt++) {
+		// Double the pixels in the line.
+		for (int i = 0; i < doubleable_width; i++) {
 			*dst-- = *src;
 			*dst-- = *src;
-			src--;
+			--src;

```

Comment:
I know this is a tiny irrelevant detail, but I noticed you changed `src--` to `--src`. Why did you do it?

Reply:
`src--` is a more complex operation semantically (and is usually implemented in terms of `--src` for more complex types). When the operand is a basic type it's a purely stylistic preference but it's nice to use operators consistently both for basic type (like pointers) and for complex types (like iterators).

Reply:
Very interesting... this must be why the iterator example I saw had ++X instead of X++. I actually changed it to X++ after copying it because I thought there was no need to ever use the prefixed version.